ios/RNMarigold.m

```diff
@@ -0,0 +1,238 @@
+
+#import "RNMarigold.h"
+#import <UserNotifications/UserNotifications.h>
+
+@interface MARMessage ()
+
+- (nullable instancetype)initWithDictionary:(nonnull NSDictionary *)dictionary;
+- (nonnull NSDictionary *)dictionary;
+
+@end
+
+@interface Marigold ()
+
+- (void)setWrapperName:(NSString *)wrapperName andVersion:(NSString *)wrapperVersion;
+
+@end
+
+@interface MARPurchase ()
+
+- (nullable instancetype)initWithDictionary:(NSDictionary *)dictionary;
+
+@end
+
+@interface RNMarigold()
+
+@property (nonatomic, strong) Marigold *marigold;
+@property (nonatomic, strong) MARMessageStream *messageStream;
+
+@end
+
+
+@implementation RNMarigold
+
+// Automatically export module as RNMarigold
+RCT_EXPORT_MODULE();
```

Comment:
This will need to be added to the new `RNEngageBySailthru` class to make sure it's available for auto linking.